lib/active_merchant/billing/gateways/bambora.rb

```diff
@@ -0,0 +1,179 @@
+require "nokogiri"
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class BamboraGateway < Gateway
+      self.live_url = 'https://www.bambora.co.nz/interface/api/dts.asmx'
+      self.test_url = 'https://demo.bambora.co.nz/interface/api/dts.asmx'
+
+      self.supported_countries = ['AU', 'NZ']
+      self.supported_cardtypes = [:visa, :master, :american_express, :diners_club, :jcb]
+
+      self.homepage_url = 'http://www.bambora.com/'
+      self.display_name = 'Bambora'
+
+      self.money_format = :cents
+
+      STANDARD_ERROR_CODE_MAPPING = {
+        "05" => STANDARD_ERROR_CODE[:card_declined],
+        "06" => STANDARD_ERROR_CODE[:processing_error],
+        "14" => STANDARD_ERROR_CODE[:invalid_number],
+        "54" => STANDARD_ERROR_CODE[:expired_card],
+      }
+
+      def initialize(options={})
+        requires!(options, :username, :password)
+        super
+      end
+
+      def purchase(money, payment, options={})
+        commit("SubmitSinglePayment") do |xml|
+          xml.Transaction do
+            xml.CustRef options[:order_id]
+            add_amount(xml, money)
+            xml.TrnType "1"
+            add_credit_card(xml, payment)
+            add_credentials(xml)
+            xml.TrnSource options[:ip]
+          end
+        end
+      end
+
+      def authorize(money, payment, options={})
+        commit("SubmitSinglePayment") do |xml|
+          xml.Transaction do
+            xml.CustRef options[:order_id]
+            add_amount(xml, money)
+            xml.TrnType "2"
+            add_credit_card(xml, payment)
+            add_credentials(xml)
+            xml.TrnSource options[:ip]
+          end
+        end
+      end
+
+      def capture(money, authorization, options={})
+        commit("SubmitSingleCapture") do |xml|
+          xml.Capture do
+            xml.Receipt authorization
+            add_amount(xml, money)
+            add_credentials(xml)
+          end
+        end
+      end
+
+      def refund(money, authorization, options={})
+        commit("SubmitSingleRefund") do |xml|
+          xml.Refund do
+            xml.Receipt authorization
+            add_amount(xml, money)
+            add_credentials(xml)
+          end
+        end
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((<CardNumber>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<CVN>)[^<]+(<))i, '\1[FILTERED]\2').
+          gsub(%r((<Password>)[^<]+(<))i, '\1[FILTERED]\2')
+      end
+
+      private
+
+      def add_credentials(xml)
+        username, account_number = options[:username].split(":")
```

Comment:
I'd prefer these being separate credentials, unless they are always presented in this concatenated form.